Subject: [PATCH] Fix DateTimeView fc issue

In one process, we can add more than one DateTimeView, but registerReceiver only once.
If register context cannot match unregister context, then FC occured.

Exception info:
java.lang.IllegalArgumentException: Receiver not registered:android.widget.DateTimeView$ReceiverInfo

For Example:
There are two packages run in systemui process,
and ActivityThread.mInitialApplication is not initialized by systemui package.
 1.add DateTimeView A(view's getApplicationContext is SystemUIService) and registerReceiver.
 2.add DateTimeView B(view's getApplicationContext is null).
 3.remove DateTimeView A.
 4.remove DateTimeView B and try to unregisterReceiver.
 5.FC occured.

Solution:
getApplicationContextIfAvailable should return unique Context in one process.

diff --git a/core/java/android/widget/DateTimeView.java b/core/java/android/widget/DateTimeView.java
index e172044..01c8f6e 100644
--- a/core/java/android/widget/DateTimeView.java
+++ b/core/java/android/widget/DateTimeView.java
@@ -451,8 +451,7 @@
         }
 
         static final Context getApplicationContextIfAvailable(Context context) {
-            final Context ac = context.getApplicationContext();
-            return ac != null ? ac : ActivityThread.currentApplication().getApplicationContext();
+            return ActivityThread.currentApplication().getApplicationContext();
         }
 
         void register(Context context) {
